lib/engines/hcktest/hcktest.rb

```diff
@@ -151,21 +151,28 @@ def configure_clients
       end
     end
 
+    def reconfigure_clients
+      configure_clients
+      @clients.each_value(&:synchronize)
+    end
+
     def configure_setup_and_synchronize
+      return reconfigure_clients unless @studio.tools.nil?
+
       @studio.configure(@platform['clients'])
       configure_clients
       @clients.each_value(&:synchronize)
       @studio.keep_snapshot
       @clients.each_value(&:keep_snapshot)
     end
 
-    def run_and_configure_setup(scope)
+    def run_and_configure_clients(scope)
       retries = 0
       begin
         scope.transaction do |tmp_scope|
-          run_studio tmp_scope
-          sleep 5 until @studio.up?
-          run_clients tmp_scope, keep_alive: true
+          run_opts = { keep_alive: true }
+          run_opts.merge!(@client_run_opts) if @client_run_opts
+          run_clients tmp_scope, run_opts
```

Comment:
```suggestion
          run_clients tmp_scope, keep_alive: true, **@client_run_opts
```

Reply:
I also want you to minimize the use of instance variables; pass `@client_run_opts` as an argument of this method instead.